framework/src/eventloop/eventloop_async.c

```diff
@@ -0,0 +1,151 @@
+/****************************************************************************
+ *
+ * Copyright 2018 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+#include <debug.h>
+#include <tinyara/config.h>
+#include <sys/types.h>
+
+#include <unistd.h>
+#include <queue.h>
+#include <semaphore.h>
+#include <libtuv/uv.h>
+#include <eventloop/eventloop.h>
+
+#include "eventloop_internal.h"
+
+struct el_async_list_s {
+	sq_queue_t async_cb_list;
+	el_async_t *handle;
+};
+typedef struct el_async_list_s el_async_list_t;
+
+struct el_async_cb_s {
+	struct el_async_cb_s *flink;
+	el_async_callback func;
+	void *cb_data;
+};
+typedef struct el_async_cb_s el_async_cb_t;
+
+static el_async_list_t g_async_list[CONFIG_MAX_TASKS];
+static sem_t el_thread_safe;
+
+#define ASYCN_CB_LIST(idx)    g_async_list[idx].async_cb_list
+#define HANDLE(idx)           g_async_list[idx].handle
+
+static int el_get_async_list_idx(void)
+{
+	int idx;
+
+	idx = PID_HASH(getpid());
+	if (idx < 0 || idx >= CONFIG_MAX_TASKS) {
+		eldbg("Invalid Parameter!\n");
+		return ERROR;
+	}
+	return idx;
+}
+
+static void el_thread_safe_async_functioncall(el_async_t *handle)
```

Comment:
argument handle is not used. Is this necessary?

Reply:
Yes, because it is callback function format of uv_async